blockstore/apps/bundles/models.py

```diff
@@ -0,0 +1,223 @@
+generated UUID4.
+
+2) Another reason we're keeping UUIDs out of the primary key is because all
+secondary keys in MySQL's InnoDB storage engine store a copy of the primary key
+entry as well. A bigint is 8 bytes, and even an optimally packed UUID is still
+16 bytes.
+
+3) Our repos generally have VARCHAR(255) everywhere because MySQL's "utf8"
+encoding uses a max of three bytes per character (note: meaning it's not really
+UTF-8), and 255 * 3 = 765 bytes. The largest size for any column in an InnoDB
+index is 767 bytes. However, if we use MySQL's "utf8mb4", a.k.a. "actually
+UTF-8", that limit goes down to 191 bytes (191 * 4 = 764). Because 191 is a
+peculiar and difficult to remember number for an API, I am somewhat arbitrarily
+using 180 as a reasonable sounding number below the limit.
+
+4) There are a number of models that are made bigint when they don't strictly
+need to be (Bundles are unlikely to grow to that size, for instance), because
+it's not *that* expensive, and migrating data once it gets that large is a pain.
+"""
+import codecs
+import uuid
+
+from django.db import models
+from django.dispatch import receiver
+import attr
+
+from .store import BundleDataStore, snapshot_created
+
+MAX_CHAR_FIELD_LENGTH = 180
+
+
+class Collection(models.Model):
+    """
+    Administrative grouping for Bundles: policy, permissions, licensing, etc.
+
+    Each Bundle belongs to exactly one Collection.
+
+    UUIDs should be treated as immutable, and external references (e.g. URLs)
+    should use them.
+
+    Target Scale: 100K rows
+    """
+    id = models.BigAutoField(primary_key=True)
+    uuid = models.UUIDField(default=uuid.uuid4, unique=True)
+    title = models.CharField(max_length=MAX_CHAR_FIELD_LENGTH, db_index=True)
+
+    def __str__(self):
+        return "{} - {}".format(self.uuid, self.title)
+
+
+class Bundle(models.Model):
+    """
+    The UUID of a Bundle should be treated as immutable, but any other part of
+    the bundle can be modified without breaking anything.
+
+    Target Scale: 100M rows
+    """
+    id = models.BigAutoField(primary_key=True)
+    uuid = models.UUIDField(default=uuid.uuid4, unique=True)
+    title = models.CharField(max_length=MAX_CHAR_FIELD_LENGTH, db_index=True)
+
+    # Comment for now just to make testing easier.
+    # collection = models.ForeignKey(
+    #     Collection, related_name="bundles", related_query_name="bundle"
+    # )
+
+    slug = models.SlugField(allow_unicode=True)  # For pretty URLs
+    description = models.TextField(max_length=10_000)
+
+    def __str__(self):
+        return "Bundle {} - {}".format(self.uuid, self.slug)
+
+
+@attr.s(frozen=True)
+class BundleVersionKey():
+    """
+    Do we really want this class at all?
+
+    BundleVersion ID serialization could be left to the view layer, since we're
+    not storing serialized BundleVersion IDs anywhere as one thing. It's mostly
+    just input shorthand.
```

Comment:
Hmm. In general I prefer to defer additive things unless there is a purpose for them. It may be needed depending on how links to other bundles are specified.

Reply:
I'll take it out for now. It's not that hard to add it back in later if we start running into it.